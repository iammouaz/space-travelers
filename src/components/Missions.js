import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Table } from 'react-bootstrap';
import Mission from './Mission';
import { getMisstions } from '../redux/missions';

const Missions = () => {
  const dispatch = useDispatch();
  const MissionsState = useSelector((state) => state.MissionReducer);
  useEffect(() => {
    if (MissionsState.length === 0) {
      dispatch(getMisstions());
    }
  }, []);
  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
        </tr>
        {MissionsState.map((mission) => (
          <Mission
            id={mission.mission_id}
            key={mission.mission_id}
            mission_name={mission.mission_name}
            description={mission.description}
          />
        ))}
      </thead>
    </Table>
  );
};

export default Missions;
