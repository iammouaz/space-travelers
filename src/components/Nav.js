import React from 'react';
import { NavLink } from 'react-router-dom';
import Logo from '../icon.png';

const Nav = () => (
  <header className="header">
    <span className="head">
      <img className="img" src={Logo} alt="logo" />
      <h1>
        {'Space Travelers\'s Hub'}
      </h1>
    </span>
    <nav className="nav">
      <NavLink to="/" className="NavLink" activeClassName="is-active" exact>
        <span className="rockets">Rockets</span>
      </NavLink>
      <NavLink
        to="/missions"
        className="NavLink"
        activeClassName="is-active"
      >
        <span className="missions ">Missions</span>
      </NavLink>
      <NavLink to="/profile" className="NavLink" activeClassName="is-active">
        <span className="profile ">My profile</span>
      </NavLink>
    </nav>
  </header>
);

export default Nav;
