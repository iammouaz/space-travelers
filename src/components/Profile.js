import React from 'react';
import { useSelector } from 'react-redux';
import { ListGroup } from 'react-bootstrap';

const Profile = () => {
  const rockets = useSelector((state) => state.rocketsRducer);
  const registeredRocket = rockets.filter((rocket) => rocket.reserved);
  const missions = useSelector((state) => state.MissionReducer);
  const registerMisson = missions.filter((mission) => mission.reserved);

  return (
    <div>
      <div className="d-flex justify-content-around">
        <div>
          <h2>My Rockets</h2>
          <ListGroup>
            {registeredRocket.map((rocket) => (
              <ListGroup.Item key={rocket.id}>{rocket.title}</ListGroup.Item>
            ))}
          </ListGroup>
        </div>
        <div>
          <h2>My Missiosn </h2>
          <ListGroup>
            {registerMisson.map((mission) => (
              <ListGroup.Item key={mission.mission_id}>
                {mission.mission_name}
              </ListGroup.Item>
            ))}
          </ListGroup>
        </div>
      </div>
    </div>
  );
};

export default Profile;
