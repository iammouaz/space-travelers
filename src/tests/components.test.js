import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import MissionReducer from '../redux/missions';

// Components
import Profile from '../components/Profile';
import Missions from '../components/Missions';
import Nav from '../components/Nav';

describe('Components Matches', () => {
  it('Mission Components Matches', () => {
    const profile = renderer
      .create(<Provider store={store}><Profile /></Provider>)
      .toJSON();
    expect(profile).toMatchSnapshot();
  });
  it('Missions Components Matches', () => {
    const missions = renderer
      .create(<Provider store={store}><Missions /></Provider>)
      .toJSON();
    expect(missions).toMatchSnapshot();
  });
  test('Check the initialState of the reducer', () => {
    expect(MissionReducer(undefined, {})).toEqual([]);
  });
});

describe('Test Nav renderisation', () => {
  test('Nav', () => {
    const nav = renderer
      .create(<BrowserRouter><Nav /></BrowserRouter>)
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
