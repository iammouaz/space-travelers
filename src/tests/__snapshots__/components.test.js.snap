// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Matches Mission Components Matches 1`] = `
<div>
  <div
    className="d-flex justify-content-around"
  >
    <div>
      <h2>
        My Rockets
      </h2>
      <div
        className="list-group"
        onKeyDown={[Function]}
      />
    </div>
    <div>
      <h2>
        My Missiosn 
      </h2>
      <div
        className="list-group"
        onKeyDown={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Components Matches Missions Components Matches 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
    </tr>
  </thead>
</table>
`;

exports[`Test Nav renderisation Nav 1`] = `
<header
  className="header"
>
  <span
    className="head"
  >
    <img
      alt="logo"
      className="img navbar-brand"
      src="icon.png"
    />
    <h1>
      Space Travelers's Hub
    </h1>
  </span>
  <nav
    className="nav justify-content-end"
  >
    <a
      aria-current="page"
      className="nav-link is-active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <span
        className="rockets"
      >
        Rockets
      </span>
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/missions"
      onClick={[Function]}
    >
      <span
        className="missions "
      >
        Missions
      </span>
    </a>
    <a
      aria-current={null}
      className="nav-link profile"
      href="/profile"
      onClick={[Function]}
    >
      <span>
        My profile
      </span>
    </a>
  </nav>
</header>
`;
